src/main/java/nextstep/subway/section/domain/Section.java

```diff
@@ -97,11 +102,11 @@ public boolean isEqualsDownStation(Section section) {
     }
 
     public boolean isEqualsUpStation(Station station) {
-        return upStation == station;
+        return upStation == station || upStation.equals(station);
```

Comment:
upStation == station 이 부분은 주소 비교를 해줄 것 같은데 혹시 이 부분이 추가되어야 할 이유가 있을까요...?

Reply:
초기화된 객체와  프록시 상태인 객체 두 객체의 값 비교를 고려해서 넣었습니다!
이 케이스에 대해서는 두 객체가 비교하는 경우를 만들지 않게 하기위해  아에 다 페치조인으로 
초기화 해놓을까 싶은 생각도 있었습니다만 위 상황이 생길 여지가 있다고 판단해서 그랬습니다
영훈님 의견은 어떠신지 여쭤봐도될까요..?

Reply:
우선 저는 이 부분에 대해서는 fetch join 혹은 더 나아가서 FetchType 을 Eager로 두는 것도 좋은 방향이라고 생각합니다. 조회해야 하는 데이터가 크지 않아 메모리를 많이 차지 하지 않을 것으로 기대되고, 데이터를 한 번에 가져올 때 추후의 로직을 수행하는데 유리할 것이라고 판단되기 때문입니다.
그런데 이 부분에 대해서 궁금한 것은 
```suggestion
        return upStation.equals(station);
```
이렇게만 두었을 때 이슈가 생길 여지가 있는지.. 하는 부분입니다. 물론 주소값이 같다면 두 객체의 값이 같을테지만, 더 정확히는 equals 함수에서 구현한 내용(id값 동일)이 현재 도메인에서의 동일성 여부를 판단하는데 더 정확한 내용이 아닐까 합니다. 정확히는 `초기화된 객체와 프록시 상태인 객체 두 객체의 값 비교를 고려해서 넣었습니다!` 부분으로 의도하신 내용을 정확히 이해하지 못한 부분도 있는 것 같습니다. 혹시 가능하다면 이 부분도 추가로 설명 부탁드리고 싶습니다!